controllers/volsync/vshandler.go

```diff
@@ -1253,10 +1253,14 @@ func (v *VSHandler) validateAndProtectSnapshot(
 
 	// Add ownerRef on snapshot pointing to the vrg - if/when the VRG gets cleaned up, then GC can cleanup the snap
 	// Add label to indicate that VolSync should not delete/cleanup this snapshot
-	err = util.NewResourceUpdater(volSnap).
-		AddOwner(v.owner, v.client.Scheme()).
-		AddLabel(VolSyncDoNotDeleteLabel, VolSyncDoNotDeleteLabelVal).
-		Update(v.ctx, v.client)
+	// Cross-namespace owner references are disallowed, so setting owner is skipped, when VRG is situated in admin
+	// namespace
+	updater := util.NewResourceUpdater(volSnap)
+	if !v.vrgInAdminNamespace {
+		updater.AddOwner(v.owner, v.client.Scheme())
```

Comment:
Who cleans up snapshots when the VRG is deleted? Each time we restore a PVC (on failover or relocate), we save the last known good snapshot for potential future use. When the VRG is deleted, all owned snapshots are garbage collected. However, with this change, the snapshots become orphaned.

Reply:
Also looking at other owner ref changes, have a similar question for the following callers:
  vshandler.go
    EnsurePVCforDirectCopy
    reconcileLocalRD
    reconcileLocalRS
    
I am assuming that createOrUpdate[RD|RS] are garbage collected by VRG as it shifts from Primary to Secondary, if an app is DR disabled, the final RD|RS garbage collection depends on the owner ref.?

(in the case of VR, which also has such an owner ref. VRG deletion, deletes the VR etc. so those are garbage collected explicitly at present).

@raghavendra-talur what are we missing if anything here?